Lib/test/libregrtest/utils.py

```diff
@@ -712,3 +713,19 @@ def get_signal_name(exitcode):
         pass
 
     return None
+
+
+ILLEGAL_XML_CHARS_RE = re.compile(
+    '['
+    '\x00-\x1F'      # ASCII control characters
+    '\uD800-\uDFFF'  # surrogate characters
+    '\uFFFE'
+    '\uFFFF'
+    ']')
+
+def _escape_xml_replace(regs):
+    code_point = ord(regs[0])
+    return f"&#{code_point};"
+
+def escape_xml(text):
```

Comment:
"Escape" is a wrong word, because "escaping" means using character references in XML, but this does not work for illegal characters. I used word "sanitize".